platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/uptime/UptimeMetrics.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.uptime;
+
+import com.swirlds.common.metrics.FunctionGauge;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.common.system.address.Address;
+import com.swirlds.common.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Metrics that track node uptime.
+ */
+class UptimeMetrics {
+
+    private static final String CATEGORY = "platform";
+
+    private final Metrics metrics;
+
+    /**
+     * A map from node to the time since the last consensus event was observed from that node.
+     */
+    private final Map<Long, RunningAverageMetric> timeSinceLastConsensusEvent = new HashMap<>();
+
+    /**
+     * A map from node to the number of rounds since the last consensus event was observed from that node.
+     */
+    private final Map<Long, RunningAverageMetric> roundsSinceLastConsensusEvent = new HashMap<>();
+
+    /**
+     * A map from node to the time since the last consensus event was observed from that node.
```

Comment:
"...time since the last ~~consensus event~~ *judge* was observed..."

Reply:
I blame the AI. Fixed.